exercises/dot-dsl/dot_dsl.py

```diff
@@ -0,0 +1,27 @@
+NODE, EDGE, ATTR = range(3)
+
+
+class Node(object):
+    def __init__(self, name, attrs={}):
+        self.name = name
+        self.attrs = attrs
+
+    def __eq__(self, other):
+        return self.name == other.name and self.attrs == other.attrs
```

Comment:
I'm wondering whether it might be better not to implement this but to instead hint that it should be implemented? I feel like rather a lot of code is being supplied pre-written, without the user having to think about how to check for equality. I think that we should either set a lower `difficulty` for this problem, or we should add a `.meta/hints.md` that lets the user know that they need to implement the `__eq__` method themself (with each method being stripped down to just `pass`).

Reply:
I was concerned that without defining `__eq__`, the tests wouldn't even pass. A lot of the test cases, such as 

```
        self.assertEqual(g.nodes, [Node("a", {"color": "green"})])
```

requires `__eq__` to be defined properly for the check to work. 

Perhaps we could say that as part of TDD, the person working on this topic should seek to first complete `__eq__` so that test cases work as the human would expect?

Reply:
Looking at it again, I think it should be ok to leave it as it is - I'm not a great fan of supplying too much code, but the definition is also fairly easy and could border on being tedious.